lang/py/src/avro/io.py

```diff
@@ -341,6 +402,59 @@ def write_double(self, datum):
     self.write(chr((bits >> 48) & 0xFF))
     self.write(chr((bits >> 56) & 0xFF))
 
+  def write_decimal_bytes(self, datum, scale):
+    """
+    Decimal in bytes are encoded as long. Since size of packed value in bytes for
+    signed long is 8, 8 bytes are written.
+    """
+    if isinstance(datum, Decimal):
+      datum = datum.to_eng_string()
+    else:
+      datum = str(datum)
+    decimal_place = datum.find('.')
+    if decimal_place == -1:
+      mantissa = 0
+    else:
+      mantissa = len(datum[decimal_place+1:])
+    if mantissa != scale:
+      raise AvroTypeException('Scale provided in schema does not match the decimal')
+    unscaled_datum = datum.replace('.', '')
+    unscaled_int_datum = long(unscaled_datum)
+    packed_bytes = STRUCT_SIGNED_LONG.pack(unscaled_int_datum)
+    self.write_long(8)
```

Comment:
The number of bytes used should depend on the decimal being written. This implementation will yield incorrect results when the precision is greater than 8 (and waste space when writing small decimals).

(There is a related issue in `read_decimal_from_bytes` above, which assumes that the length can only be 2, 4, or 8.)